plugins/producer_plugin/producer_plugin.cpp

```diff
@@ -381,12 +383,22 @@ class producer_plugin_impl : public std::enable_shared_from_this<producer_plugin
             throw;
          };
 
+         // Notes:
+         // - as soon as block is accepted via `push_block` a signal is emitted.
+         // - may be able to just overload `push_block` to _verify_ the block if blockvault is currently active; will have to just pass noop lambda:
+         //   - this is no longer planned.
          try {
             chain.push_block( bsf, [this]( const branch_type& forked_branch ) {
                _unapplied_transactions.add_forked( forked_branch );
             }, [this]( const transaction_id_type& id ) {
                return _unapplied_transactions.get_trx( id );
             } );
+            if ( blockvault_plug ) {
+               uint32_t lib{chain.head_block_state()->dpos_irreversible_blocknum};
+               eosio::chain::signed_block_ptr block{chain.head_block_state()->block};
```

Comment:
No reason to make a copy of the `signed_block_ptr`, just pass it directly to `append_external_block`. Again `head_block_state` is not necessarily the block you just pushed.

Reply:
Changed the code to just use the `block_state_ptr` returned from `push_block` in place of `lib` and `block`.